pallets/identity/src/claims.rs

```diff
@@ -481,6 +481,19 @@ impl<T: Config> Module<T> {
         // Sender has to be part of CDDProviders
         Self::ensure_authorized_cdd_provider(cdd_did)?;
 
+        let record = <DidRecords<T>>::get(cdd_did);
+        let cost =
+            secondary_keys
+                .iter()
+                .try_fold(0usize, |cost, auth| -> Result<usize, DispatchError> {
+                    //Check permissions limits
+                    Self::ensure_perms_length_limited(&auth.permissions)?;
+                    Ok(cost.saturating_add(auth.permissions.complexity()))
+                });
+
+        // Check secondary keys limits
+        Self::ensure_secondary_keys_limited(&record, secondary_keys.len(), cost.unwrap())?;
```

Comment:
`cost.unwrap()` again with the unwrap, this returns the error result into a panic.  Errors need to be correctly handled.